### ISSUE TITLE
Medusa effected by bug CASSANDRA-17581

### ISSUE THREAD

**Issue by sandeepmallik** [thelastpickle/cassandra-medusa#472]
Hi,

We are using Cassandra 3.0.15 and Java 1.8.0_332. It seems we are effected by bug - https://issues.apache.org/jira/browse/CASSANDRA-17581. 

We can't upgrade to latest Cassandra version to overcome this. I tried alias (alias nodetool='nodetool -Dcom.sun.jndi.rmiURLParsing=legacy') and that didn't work either. 

Is there a patch or way around to overcome this in medusa?

[2022-05-06 03:45:52,475] INFO: Registered backup id test1234
[2022-05-06 03:45:52,475] INFO: Monitoring provider is noop
[2022-05-06 03:45:53,352] INFO: Starting backup using Stagger: None Mode: full Name: test1234
[2022-05-06 03:45:53,352] INFO: Updated from existing status: -1 to new status: 0 for backup id: test1234 
[2022-05-06 03:45:53,353] INFO: Saving tokenmap and schema
[2022-05-06 03:45:54,701] INFO: Node 172.20.181.86 does not have latest backup
[2022-05-06 03:45:54,701] INFO: Creating snapshot
**nodetool: Failed to connect to '127.0.0.1:7199' - URISyntaxException: 'Malformed IPv6 address at index 7: rmi://[127.0.0.1]:7199'.**
[2022-05-06 03:45:55,600] ERROR: Issue occurred inside handle_backup Name: test1234 Error: Command '['nodetool', '-u', 'cassandra', '-pw', 'cassandra', 'snapshot', '-t', 'medusa-test1234']' returned non-zero exit status 1.
[2022-05-06 03:45:55,600] INFO: Updated from existing status: 0 to new status: 2 for backup id: test1234 
[2022-05-06 03:45:55,600] ERROR: Error occurred during backup: Command '['nodetool', '-u', 'cassandra', '-pw', 'cassandra', 'snapshot', '-t', 'medusa-test1234']' returned non-zero exit status 1.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/medusa/backup_node.py", line 196, in handle_backup
    info = start_backup(storage, node_backup, cassandra, differential_mode, stagger_time, start, mode,
  File "/usr/local/lib/python3.9/site-packages/medusa/backup_node.py", line 249, in start_backup
    num_files, node_backup_cache = do_backup(
  File "/usr/local/lib/python3.9/site-packages/medusa/backup_node.py", line 294, in do_backup
    with cassandra.create_snapshot(backup_name) as snapshot:
  File "/usr/local/lib/python3.9/site-packages/medusa/cassandra_utils.py", line 449, in create_snapshot
    self.snapshot_service.create_snapshot(tag=tag)
  File "/usr/local/lib/python3.9/site-packages/medusa/service/snapshot/nodetool_snapshot_service.py", line 32, in create_snapshot
    subprocess.check_call(cmd, stdout=subprocess.DEVNULL, universal_newlines=True)
  File "/usr/local/lib64/python3.9/site-packages/gevent/subprocess.py", line 332, in check_call
    raise CalledProcessError(retcode, cmd) # pylint:disable=undefined-variable
subprocess.CalledProcessError: Command '['nodetool', '-u', 'cassandra', '-pw', 'cassandra', 'snapshot', '-t', 'medusa-test1234']' returned non-zero exit status 1.

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-1489) by [Unito](https://www.unito.io)
┆friendlyId: K8SSAND-1489
┆priority: Medium

**Comment by adejanovski** [comment]
Hi @sandeepmallik, I've created a PR to allow adding flags in the nodetool commands and set `-Dcom.sun.jndi.rmiURLParsing=legacy` as default flag.

Could you give that #473 PR a test ride and tell me if it works as expected?

Thanks

**Comment by abonacin** [comment]
Hi @adejanovski.
We tested this version:
```
[cassandra@ip-20-0-1-224 ~]$ medusa --version
0.12.3.dev0
```

```
[cassandra@ip-20-0-1-224 ~]$ java -version
openjdk version "1.8.0_332"
OpenJDK Runtime Environment (build 1.8.0_332-b09)
OpenJDK 64-Bit Server VM (build 25.332-b09, mixed mode)
[cassandra@ip-20-0-1-224 ~]$
```

We can see with debug that looks good now:
```
[cassandra@ip-20-0-1-224 ~]$ medusa -vvv  backup --backup-name=backup_1
...
[2022-05-06 19:00:34,814] DEBUG: [Storage] Getting object 20.0.1.224/None/meta/incremental
[2022-05-06 19:00:34,819] DEBUG: https://s3.amazonaws.com:443 "HEAD /bucket-backup HTTP/1.1" 200 0
[2022-05-06 19:00:34,833] DEBUG: https://s3.amazonaws.com:443 "HEAD /bucket-backup/20.0.1.224/None/meta/incremental HTTP/1.1" 404 0
[2022-05-06 19:00:34,833] INFO: Node 20.0.1.224 does not have latest backup
[2022-05-06 19:00:34,833] INFO: Creating snapshot
[2022-05-06 19:00:34,835] DEBUG: Executing: nodetool -Dcom.sun.jndi.rmiURLParsing=legacy -u cassandra -pw cassandra snapshot -t medusa-backup_1
[2022-05-06 19:00:39,149] DEBUG: Backing up <medusa.cassandra_utils.SnapshotPath object at 0x7fdc5c5d0b20>
[2022-05-06 19:00:39,150] DEBUG: destination path: 20.0.1.224/data/system/IndexInfo-9f5c6374d48532299a0a5094af9ad1e3
...
[2022-05-06 19:01:06,879] INFO: Backup done
[2022-05-06 19:01:06,879] INFO: - Started: 2022-05-06 19:00:30
                        - Started extracting data: 2022-05-06 19:00:34
                        - Finished: 2022-05-06 19:01:06
[2022-05-06 19:01:06,879] INFO: - Real duration: 0:00:32.146270 (excludes time waiting for other nodes)
[2022-05-06 19:01:06,880] INFO: - 216 files, 163.85 KB
[2022-05-06 19:01:06,880] INFO: - 216 files copied from host
[2022-05-06 19:01:06,880] DEBUG: Emitting metrics
[2022-05-06 19:01:06,881] DEBUG: Done emitting metrics
[2022-05-06 19:01:06,881] INFO: Updated from existing status: 0 to new status: 1 for backup id: backup_1
[2022-05-06 19:01:06,881] DEBUG: Done with backup, returning backup result information
...
[cassandra@ip-20-0-1-224 ~]$ medusa list-backups
backup_1 (started: 2022-05-06 19:00:34, finished: 2022-05-06 19:34:38)
[cassandra@ip-20-0-1-224 ~]$
```

Thank you.

**Comment by adejanovski** [comment]
Fix released as part of v0.12.3
